src/blueprints/tv2/factories/tv2-graphics-action-factory.ts

```diff
@@ -134,5 +148,162 @@ export class Tv2GraphicsActionFactory {
+      case Tv2GraphicsType.VIZ:
+      default: return this.createVizFullscreen(blueprintConfiguration, actionManifests)
+    }
+  }
+
+  private createVizFullscreen(blueprintConfiguration: Tv2BlueprintConfiguration, actionManifests: Tv2GraphicActionManifest[]): PartAction[] {
+    return actionManifests.map((manifest) => this.createVizGraphicsActionFromManifest(
+      blueprintConfiguration,
+      manifest
+    ))
+  }
+
+  private createVizGraphicsActionFromManifest(blueprintConfiguration: Tv2BlueprintConfiguration, manifest: Tv2GraphicActionManifest): PartAction {
+    const target: Tv2GraphicsTarget = Tv2GraphicsTarget.FULL
+    const fullGraphicPiece: PieceInterface = this.createVizFullGraphicsPiece(blueprintConfiguration, manifest)
+    const partInterface: PartInterface = this.createGraphicsPartInterface({
+      id: `${this.getGraphicTargetAsCamelString(target)}Part`, // Todo: make id unique
+      name: `${this.getGraphicTargetAsCamelString(target)} ${manifest.userData.name}`,
+      inTransition: {
+        keepPreviousPartAliveDuration: blueprintConfiguration.studio.VizPilotGraphics.KeepAliveDuration,
+        delayPiecesDuration: 0
+      },
+    })
+    return {
+      id: '',
+      name: '',
+      type: PartActionType.INSERT_PART_AS_NEXT,
+      data: {
+        partInterface: partInterface,
+        pieceInterfaces: [ fullGraphicPiece ]
+      }
+    }
+  }
+
+  private createVizFullGraphicsPiece(blueprintConfiguration: Tv2BlueprintConfiguration, manifest: Tv2GraphicActionManifest): PieceInterface {
+    return this.createGraphicsPieceInterface({
+      id: 'fullGraphicPiece', // Todo: make id unique
+      name: manifest.userData.name,
+      preRollDuration: blueprintConfiguration.studio.VizPilotGraphics.PrerollDuration,
+      pieceLifespan: this.findInfiniteModeFromConfig(blueprintConfiguration, manifest),
+      layer: Tv2SourceLayer.PILOT_GRAPHIC,
+      content: this.createVizFullGraphicsPieceContent(blueprintConfiguration, manifest)
+    })
+  }
+
+  private createGraphicsPartInterface(partInterfaceWithRequiredValues: Pick<PartInterface, 'id' | 'name'> & Partial<PartInterface>): PartInterface {
+    return {
+      disableNextInTransition: false,
+      expectedDuration: 0,
+      inTransition: {
+        keepPreviousPartAliveDuration: 0,
+        delayPiecesDuration: 0
+      },
+      isNext: true,
+      isOnAir: false,
+      isPlanned: false,
+      outTransition: {
+        keepAliveDuration: 0
+      },
+      pieces: [],
+      rank: 0,
+      segmentId: '',
+      ...partInterfaceWithRequiredValues
+    }
+  }
+
+  private getGraphicTargetAsCamelString(target: Tv2GraphicsTarget): string {
+    return target.toString().charAt(0) + target.toString().substring(1).toLowerCase()
+  }
+
+  private findInfiniteModeFromConfig(blueprintConfiguration: Tv2BlueprintConfiguration, manifest: Tv2GraphicActionManifest): PieceLifespan {
```

Comment:
You are finding the `PieceLifespan`. Instead of inventing a new term for that just call it `findPieceLifespan()`.
A good tip: Look at the return type of the method to determine what to call it.

Reply:
Has been renamed to `findPieceLifespan` now. Original name was brought over from old Blueprints.